### PR TITLE
refactor(requests): own request iteration

### PR DESCRIPTION
**Pull request by paulfalgout**
Related #241

## Linked issue

- Related #241

## What

- Replace the Requests mixin's complete Underscore import with captured own-key snapshots and indexed accumulator loops.
- Preserve request registration, removal, dispatch, aggregation, default-handler, once-handler, logging, and error-cutoff semantics.
- Reorganize and expand per-method request contract coverage, then regenerate affected main distributions.

## Why

Requests only needs a narrow set of map and dense-list operations. Owning those operations directly removes another production dependency edge, avoids callback binding and Array iterator overhead, and makes the message-dispatch contract legible to agents.

## Agent-development failure addressed

- Makes request-map key scope, snapshot timing, recursion order, collision behavior, receiver/arguments forwarding, and partial-mutation boundaries explicit.
- Prevents agents from replacing the loops with native callbacks or shared helpers that change getter/trap ordering or accumulator assignment timing.
- Groups behavior under the public method responsible for it so failures route agents to the correct implementation boundary.

## Public behavior contract

- Canonical behavior after this PR: request maps use captured own enumerable string-key snapshots in property order; request argument lists are reduced densely by captured length; successful registration preserves the existing registry identity; failures retain only mutations completed before the throw.
- `stopReplying` preserves name/callback/context truthiness and matching, own-key lookup-before-get behavior, snapshot deletion order, and return identity.
- object-form and whitespace-form `request` calls preserve recursive dispatch order, last-write collision behavior, safe built-in and `__proto__` results, direct/default handler priority, receiver/arguments forwarding, once removal, tuned logging, and error cutoff.
- Superseded behavior removed: direct Underscore `each`, `keys`, and `reduce` use in Requests.
- Compatibility exception and removal condition: none; the owned implementation preserves the prior observable contract.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: direct production Underscore importers drop from 8 to 7. Exact-base Brotli deltas are ESM +117 B, CJS +127 B, UMD +129 B, and minified +118 B; cumulative measured artifacts grow 491 B.
- Growth approval: minified UMD grows about 1.014%, just over the 1% exact-base approval threshold, so this PR requires an allowed maintainer comment approving this exact head before merge.
- Startup/hot path: one `Object.keys` reference is captured at module load; request operations use indexed loops without per-item bound callbacks or Array iterator machinery.
- Allocations/retention: map operations keep their existing key snapshots and response objects while removing reducer/iterator callback allocations.

## Scope

- Affects only the Requests mixin and generated main artifacts.
- Does not change Events, Radio channel construction, request debug configuration, entity binding helpers, or other Underscore consumers.

## Deployment Impact

- No website, package, tag, release, or deployment is performed by this PR.
- No consumer redeploy is required beyond adopting a future Marionette package containing this change.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm ci --no-audit --no-fund
npm exec vitest -- run test/unit/requests.spec.js test/unit/radio.spec.js test/unit/common/bind-request.spec.js test/unit/mixins/radio.spec.js --reporter=dot
npm run lint:ci
git diff --check
npm run build
npm test -- --reporter=dot
npm run test:source
npm run test:dist
npm run test:fixtures
npm run docs:check
npm run check:browser-profile
npm run check:diagnostics
npm run test:performance
npm run size
npm run performance:timing
npm run check:dist
```

Results: focused tests passed 95/95 across four files; full tests passed 1,534/1,534 across 81 files; source, distribution, packed CJS/ESM/types/Vite fixtures, documentation, browser profile, diagnostics, lint, build, size/resource, 92 performance-contract, hosted timing, post-commit distribution, and diff validation passed. Independent exact-head review passed 35/35 Requests tests with zero findings.

## Package and browser impact

- Entry points or install fixtures affected: main ESM, CJS, UMD, and minified artifacts are regenerated; packed CJS/ESM/type/Vite fixtures pass.
- Browser contracts affected: none; the implementation uses baseline object keys, indexed arrays, regular expressions, and property operations.
- Production/dev/test module-graph impact: Requests stops importing Underscore; other production consumers still retain it.

## Rollback or deprecation

- Revert this single commit and regenerate the main distribution artifacts; no data migration or compatibility bridge is involved.

## Review notes

- Indexed accumulator loops intentionally avoid native `Array.prototype.reduce`, iterator lookup, and bound reducer callbacks while preserving dense captured-length behavior.
- Object-form request aggregation continues to merge split-name results safely and uses safe property writes for literal built-in and `__proto__` request names.

---
## Summary by cubic
Replaces the Requests mixin's `each`, `keys`, and `reduce` imports from `underscore` with captured own-key snapshots and indexed accumulator loops. Request registration, removal, dispatch, aggregation, default-handler, once-handler, logging, and error-cutoff behavior is unchanged.

**Refactors**
- Expanded per-method request contract tests and isolated request fixtures.
- Regenerated `dist/` artifacts; minified UMD grows about 1.014%, past the 1% exact-base approval threshold, so a maintainer comment approving this head is required before merging.
- Reverting the commit and regenerating main artifacts rolls back cleanly; no migration or compatibility bridge is involved.

<sup>Written for commit 96334f56b844e4494890968629ae85dac5cf94b9. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/286?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved reliability when handling request and reply registries, including edge cases involving special keys, inherited properties, and primitive values.
  * Preserved existing behavior for duplicate replies, nested requests, split request names, and context handling.

* **Tests**
  * Expanded coverage for request/reply behavior, error propagation, registry updates, and special-case inputs.

### CODE REVIEW

**Comment by paulfalgout** [comment]
## Code-review approval — exact head `e12640b3`

**Approved on the code.** `Bundle size` is red only on a growth approval, not a violation: cumulative 58,639 against 75,000, and three of four artifacts sit under the threshold. The minified UMD crossed it by a hair:

```
Main ES module      +117 B (+0.77%)   Not required
Main CommonJS       +127 B (+0.83%)   Not required
Main UMD            +129 B (+0.83%)   Not required
Main minified UMD   +118 B (+1.01%)   Required   <- 0.01% over
```

So this needs an exact-head maintainer growth-approval comment naming `e12640b3`, and nothing else.

The translation reads faithfully throughout. `getKeys` reproduces `_.keys` with the same object-or-function guard used in #284's `setAttributes`, so non-objects yield `[]` rather than `Object.keys`' indexed characters. In `stopReducer` the callback's `return` becomes `continue`, which is the same control flow. And the `reduce(…, fn.bind(this), seed)` calls become explicit loops that thread the accumulator identically — same seeding, same reassignment, minus the per-call `.bind` allocation.

Both `request()` branches keep their distinction intact: `assignOwn` when the key contains a splitter and `setProperty` otherwise, so a `__proto__` key still lands as an own entry rather than mutating the reply object.

**Separately, I owe you a correction on this file — see my note on #267.** The `stopReplying({name: fn})` behavior I flagged there as a silent no-op isn't one; reading `stopReducer`'s guard here made that clear. Details are on that thread so this one stays about the PR.

Still source-only review; `node_modules` remains empty.

<sub>Reviewed with Claude Code.</sub>

File: test/unit/requests.spec.js [hunk 1]
```diff
@@ -10,145 +14,683 @@ describe('Requests', function() {
...
+      expect(Object.keys(splitThenDirect)).to.deep.equal(['first', 'second']);
+      expect(splitThenDirect).to.deep.equal({ first: 'direct', second: 'split' });
+    });
+
+    it('builds request result maps with safe own collision keys', function() {
+      const protoValue = { safe: true };
+      const requestMap = { constructor: 'argument', toString: 'argument' };
+      Object.defineProperty(requestMap, '__proto__', {
+        enumerable: true,
+        value: 'argument'
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/requests.spec.js:560]: Not changing this because Object.defineProperty creates an own __proto__ data property here; it does not assign or validate an object prototype.

**Comment by paulfalgout** [comment]
```json
{
  "schemaVersion": 1,
  "headSha": "e12640b35823a2b4f1cbbadda796d9680446b7f6",
  "issueUrl": "https://github.com/marionettejs/marionette/issues/127",
  "approvedPaths": [
    "dist/marionette.min.js"
  ],
  "evidenceUrls": [
    "https://github.com/marionettejs/marionette/issues/127#issuecomment-5466153196"
  ]
}
```

File: test/unit/requests.spec.js [hunk 2]
```diff
@@ -10,145 +14,683 @@ describe('Requests', function() {
...
+        },
+        getOwnPropertyDescriptor(object, key) {
+          trace.push(`descriptor:${key}`);
+          return Reflect.getOwnPropertyDescriptor(object, key);
+        },
+        get(object, key, receiver) {
+          trace.push(`get:${key}`);
+          if (key === 'first') {
+            delete object.second;
+            object.added = handler(callback, this.requests);
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/requests.spec.js]: Handled in 96334f5 by capturing the outer Requests context separately from the Proxy receiver and asserting the added handler retains it.

File: test/unit/requests.spec.js [hunk 3]
```diff
@@ -10,145 +14,683 @@ describe('Requests', function() {
...
+      }
+    });
 
-  it('ignores inherited named and default handlers', function() {
-    const namedHandler = this.sinon.stub();
-    const defaultHandler = this.sinon.stub();
-    this.requests._rdRequests = Object.create({
-      default: { callback: defaultHandler, context: this.requests },
-      inherited: { callback: namedHandler, context: this.requests }
+    it('does not assign a fresh replyOnce registry when wrapper setup throws', function() {
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/requests.spec.js]: Handled in 96334f5 by moving the registry-failure case into the sole #replyOnce suite.

**Comment by paulfalgout** [comment]
```json
{
  "schemaVersion": 1,
  "headSha": "96334f56b844e4494890968629ae85dac5cf94b9",
  "issueUrl": "https://github.com/marionettejs/marionette/issues/127",
  "approvedPaths": [
    "dist/marionette.min.js"
  ],
  "evidenceUrls": [
    "https://github.com/marionettejs/marionette/issues/127#issuecomment-5466153196"
  ]
}
```

**Comment by paulfalgout** [comment]
**Approval carries** to `96334f56` (the SHA I fetched). Tests-only — `dist/`, `mixins/`, `modules/`, and `utils/` are byte-identical to `e12640b3`. `Bundle size` is still the only red check, still on the growth approval rather than a violation.

Moving the wrapper-setup-throws test from `#reply` into `#replyOnce` puts it where it belongs — it exercises `replyOnce`'s `stopReplying` lookup, so a failure there was previously reported against the wrong method.

The test itself is a good one to have kept intact through the rewrite: it asserts that when wrapper setup throws, `_rdRequests` is **not** left assigned as an own property. This PR changed `this._rdRequests = registerReplies(this, eventArgs, this._rdRequests || {}, false)` from the `reduce` form, and the atomicity depends on the assignment happening only after the call returns — which the loop preserves, but which nothing else in the suite would catch if a future refactor seeded the registry before iterating.

Still outstanding: an exact-head maintainer growth-approval comment naming `96334f56` — note the head moved, so an approval issued against `e12640b3` no longer applies.

<sub>Reviewed with Claude Code.</sub>
